src/evaluators/FunctionExpression.js

```diff
@@ -52,6 +52,7 @@ export default function(
 
       // 7. Let closure be GeneratorFunctionCreate(Normal, FormalParameters, GeneratorBody, funcEnv, strict).
       let closure = GeneratorFunctionCreate(realm, "normal", ast.params, ast.body, funcEnv, strict);
+      invariant(closure instanceof ECMAScriptFunctionValue);
```

Comment:
Make the return type of GeneratorFunctionCreate be ECMAScriptFunctionValue, and avoid such invariants. This will require changing some more signatures, e.g. of FunctionAllocate.